src/libraries/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests.csproj

```diff
@@ -1,12 +1,16 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <DefineConstants Condition="'$(TargetsWindows)' == 'true'">$(DefineConstants);TargetsWindows</DefineConstants>
     <IncludeRemoteExecutor>true</IncludeRemoteExecutor>
     <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-Unix;$(NetCoreAppCurrent)-Browser</TargetFrameworks>
     <IgnoreForCI Condition="'$(TargetOS)' == 'Browser'">true</IgnoreForCI>
     <EnableDllImportGenerator>true</EnableDllImportGenerator>
   </PropertyGroup>
+  <!-- DesignTimeBuild requires all the TargetFramework Derived Properties to not be present in the first property group. -->
+  <PropertyGroup>
+    <TargetPlatformIdentifier>$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)'))</TargetPlatformIdentifier>
+    <DefineConstants Condition="'$(TargetPlatformIdentifier)' == 'windows'">$(DefineConstants);TargetsWindows</DefineConstants>
```

Comment:
Existing issue (#39174), but it would be a nice cleanup if we consolidate `TARGET_<platform/architecture>` constants.
In most places, we use `;TARGET_WINDOWS` constant, then in some places we use `;TargetsWindows`.

Perhaps we can move https://github.com/dotnet/runtime/blob/ce15dc7d1416ce8ebc4cffe7ea8c51b071f2fd3c/src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems#L30-L41 and https://github.com/dotnet/runtime/blob/ce15dc7d1416ce8ebc4cffe7ea8c51b071f2fd3c/src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems#L30-L41 into a top-level props and then consistently use them in managed code throughout the repo.

Reply:
Agreed that consolidating constants would be a good idea but I wouldn't want to do that as part of this PR. I have very limited time atm (I'm still on holiday) and want to focus on a few specific goals.

The linked issue doesn't seem to complex. Maybe the SDK already offers defines for platforms?

Reply:
Yup, it's an existing issue, not for this PR. Just wanted to highlight it since PR is touching this area. :)

> Maybe the SDK already offers defines for platforms?

When new platforms are added, it's easier to add an entry in the SDK hook (`Targets<NewPlatformName>` property in `eng/BeforeTargetFrameworkInference.targets `) in runtime repo, before full SDK support is available. I was thinking that we could define `TARGET_` constant, in the same vein, in top level props to avoid updating proj files individually. That constant will be unused in most projects (as generally partial classes are preferred over `#if TARGET_`), but most constants defined by SDK are also unused, so it's just good for consistency.

Reply:
The SDK already sets platform provides platform defines and it doesn't differentiate between well known platforms and new/unknown ones: https://github.com/dotnet/sdk/blob/997376c6bc798af22633a8eb36074a0acec8a200/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets#L193-L204.

It sets the define based of the TargetPlatformIdentifier and TargetPlatformVersion, which we provide for the custom, unknown platforms. The only thing that currently can't be expressed is inheritance, as the SDK doesn't know that `#if UNIX` is an applicable constant not just for the `UNIX` TargetPlatformIdentifier but also for i.e. the `LINUX` TargetPlatformIdentifier. That said, expressing inheritance isn't the common case, often it's a direct match i.e. `net7.0-android` -> `#if ANDROID`.

I would start with cleaning up all custom platform defines which aren't part to something, i.e. `Android`, `iOS`, `tvOS`, etc.